spirv_msl.cpp

```diff
@@ -2625,6 +2732,47 @@ uint32_t CompilerMSL::add_interface_block(StorageClass storage, bool patch)
 		add_variable_to_interface_block(storage, ib_var_ref, ib_type, *p_var, meta);
 	}
 
+	if (get_execution_model() == ExecutionModelTessellationControl && msl_options.multi_patch_workgroup &&
+	    storage == StorageClassInput)
+	{
+		// For tessellation control inputs, add all outputs from the vertex shader to ensure
```

Comment:
Do we ensure that the tessellation control inputs are declared in an expected order?

Reply:
Since vertex shader uses attributes to write varying, I don't see how we guarantee that tessellation inputs are declared at the expected Offset.

Reply:
They should be sorted by location in both shaders.